### DISCUSSION TITLE
cloud_firestore flutter bug

### DISCUSSION THREAD

**Discussion by ammar313-cs** [community/community#75811] [Programming Help]
### Body

Hi Github community, I am having the following while compiling my flutter project:

FAILURE: Build failed with an exception.

*** What went wrong:**
A problem occurred configuring project ‘:cloud_firestore’.
> Failed to create Jar file C:\Users<username>.gradle\caches\jars-8\fef84f4f98be9f93b0b593ccb1e3e207\lint-model-31.1.2.jar.
> Failed to notify project evaluation listener.
** > Could not get unknown property ‘android’ for project ‘:cloud_firestore’ of type org.gradle.api.Project.**
** > Could not find method implementation() for arguments [project ‘:firebase_core’] on object of type org.gradle.api.internal.artifacts.dsl.dependencies.DefaultDependencyHandler.**
** > Could not get unknown property ‘android’ for project ‘:cloud_firestore’ of type org.gradle.api.Project.**

*** Try:**
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

*** Get more help at https://help.gradle.org/**

BUILD FAILED in 3s
Running Gradle task ‘assembleDebug’… 4.8s
Exception: Gradle task assembleDebug failed with exit code 1

It started after i added firebase_storage dependency and this issue is now persisting .
I have tried updating plugins and sdks from android studio as well puting the latest versions of dependencies in pubsec.yaml file but it is of no avail. I have repaired pub cache as well and gone through every possible solution available on stackoverflow and github community but the error has not been resolved.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ammar313-cs** [comment]
Hey, github community. After encountering this problem myself and seeing on the internet that many others have also faced a similar issue here is how I solved it, you have to follow these basic steps and you are good to go:
1)Update your sdks in android studio
2)Apply the latest versions of dependencies in your project pubsec.yaml file 
3)Delete your ~/.graddle/cache folder in its entirety 
4)Run flutter stable channel command(I ran it in in my project folder)
6)Run flutter upgrade(ran it in project folder)
7)Run the flutter run command and hopefully, it will also solve your problem.

The process may take time so patience is the key. My problem started after I added a firebase_storage dependency to my project and this error usually appears after you add a firebase dependency.

Happy coding to you all, godspeed.

**Comment by iambhabha** [comment]
The error you're encountering seems to be related to the configuration of the cloud_firestore project and an issue with the firebase_core dependency. Here are a few steps you can take to troubleshoot and potentially resolve the issue:

Check Dependency Versions:

Ensure that you are using compatible versions of cloud_firestore and firebase_core in your pubspec.yaml file. They should be using versions that are compatible with each other.
Gradle Version:

Make sure you are using a compatible version of Gradle. Check the gradle-wrapper.properties file in your project for the Gradle version. You can also try upgrading Gradle to the latest version compatible with your Flutter version.
Update Flutter and Dart SDKs:

Ensure you are using the latest stable version of Flutter and Dart SDK. You can run flutter upgrade and flutter pub upgrade to update to the latest versions.
Clean and Rebuild:

Run flutter clean to delete the build cache, and then try rebuilding your project with flutter build or flutter run.
Check Project Structure:

Ensure that the project structure is correct. Make sure that the android folder is located at the root of your Flutter project.
Check for Duplicate Dependencies:

Ensure that you don't have conflicting or duplicate dependencies in your pubspec.yaml. Sometimes, different packages may bring in conflicting versions of the same transitive dependency.
Check Firebase Storage Documentation:

Refer to the official documentation of firebase_storage and cloud_firestore for any specific configuration or versioning requirements.
Check Gradle Configuration:

Inspect the build.gradle files of both the Flutter project and the Firebase-related modules. Ensure that the Gradle configuration is correct.
Check for Flutter and Dart SDK Path:

Make sure that the paths to Flutter and Dart SDKs are correctly set in your IDE (such as VS Code or Android Studio).
Consult Flutter Community:

If the issue persists, consider reaching out to the Flutter community on GitHub or other forums. Someone might have encountered a similar issue and could provide insights or solutions.
Remember to back up your project before making significant changes. If the issue persists, providing more details about your pubspec.yaml, relevant portions of your Gradle files, and any other relevant configurations would be helpful for further assistance.
